docs/source/user-guide/sql/window_functions.md

```diff
@@ -0,0 +1,204 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either expressioness or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Window Functions
+
+Window functions calculate a value for a set of rows within a result set.
+
+## Aggregate functions
+
+All [aggregate functions](aggregate_functions.md) can be used as window functions.
+
+Examples:
```

Comment:
Thanks for picking this @toppyy 
i would also document examples with `range/rows between leading/preceding/current row`

Reply:
👍 

There are a bunch of example queries of the syntax supported in the tests here: https://github.com/apache/arrow-datafusion/blob/dd3a003c1ca4e2109f33277d13f2b0b2fa500337/datafusion/core/tests/sqllogictests/test_files/window.slt#L1143-L1151

Reply:
Good point, added a couple of examples :thumbsup: